src/modules/esl-carousel/core/esl-carousel.ts

```diff
@@ -0,0 +1,302 @@
+  @attr({defaultValue: 'default'}) public type: string;
+  /** Marker to enable loop mode for a carousel (`true` by default). Supports {@link ESLMediaRuleList} syntax */
+  @attr({defaultValue: 'false'}) public loop: string;
+  /** Count of slides to show on the screen (`1` by default). Supports {@link ESLMediaRuleList} syntax */
+  @attr({defaultValue: '1'}) public count: string;
+  /** Orientation of the carousel (`horizontal` by default). Supports {@link ESLMediaRuleList} syntax */
+  @attr({defaultValue: 'false'}) public vertical: string;
+
+  /** true if carousel is in process of animating */
+  @boolAttr({readonly: true}) public animating: boolean;
+
+  /** Renderer type {@link ESLMediaRuleList} instance */
+  @memoize()
+  public get typeRule(): ESLMediaRuleList<string> {
+    return ESLMediaRuleList.parse(this.media, this.type);
+  }
+  /** Loop marker {@link ESLMediaRuleList} instance */
+  @memoize()
+  public get loopRule(): ESLMediaRuleList<boolean> {
+    return ESLMediaRuleList.parse(this.media, this.loop, parseBoolean);
+  }
+  /** Count of visible slides {@link ESLMediaRuleList} instance */
+  @memoize()
+  public get countRule(): ESLMediaRuleList<number> {
+    return ESLMediaRuleList.parse(this.media, this.count, parseInt);
+  }
+  /** Orientation of the carousel {@link ESLMediaRuleList} instance */
+  @memoize()
+  public get verticalRule(): ESLMediaRuleList<boolean> {
+    return ESLMediaRuleList.parse(this.media, this.vertical, parseBoolean);
+  }
+
+  /** Carousel instance current {@link ESLCarouselStaticState} */
+  public get config(): ESLCarouselStaticState {
+    return this.renderer.config;
+  }
+
+  /** Carousel instance configured {@link ESLCarouselStaticState} */
+  public get configCurrent(): ESLCarouselConfig {
+    return {
+      type: this.typeRule.value || 'default',
+      size: this.$slides.length,
+      count: this.countRule.value || 1,
+      loop: !!this.loopRule.value,
+      vertical: !!this.verticalRule.value
+    };
+  }
+
+  /** Carousel instance current {@link ESLCarouselState} */
+  public get state(): ESLCarouselState {
+    return Object.assign({}, this.renderer.config, {
+      activeIndex: this.activeIndex
+    });
+  }
+
+  /** @returns currently active renderer */
+  @memoize()
+  public get renderer(): ESLCarouselRenderer {
+    return ESLCarouselRenderer.registry.create(this, this.configCurrent);
+  }
+
+  @ready
+  protected override connectedCallback(): void {
+    super.connectedCallback();
+    this.update();
+    this.updateA11y();
+  }
+
+  protected override attributeChangedCallback(attrName: string, oldVal: string, newVal: string): void {
+    if (!this.connected) return;
+    memoize.clear(this, `${attrName}Rule`);
+    this.update();
+  }
+
+  protected override disconnectedCallback(): void {
+    super.disconnectedCallback();
+    memoize.clear(this, ['$container', '$slides', '$slidesArea']);
```

Comment:
Should we clear all `*Rule` memoizations?

Reply:
You are correct, thanks!